system/ping/ping.c

```diff
@@ -242,10 +242,17 @@ static void ping_result(FAR const struct ping_result_s *result)
                    result->extra / USEC_PER_MSEC);
             if (result->nreplies > 0)
               {
-                long avg = priv->tsum / result->nreplies;
-                long long tempnum = priv->tsum2 / result->nreplies -
+                long tmdev = 0;
+                long avg = 0;
+                long long tempnum = 0;
+
+                if (priv->tsum > 0)
```

Comment:
But, I don't see tsum used as divider. BTW, should we modify ping6 too?

Reply:
tsum is used to calculate the other variables, if it's 0 all of the rest will be 0 including the one passed to `ub32sqrtub16` which will end up crashing due to a division by zero.

Reply:
Regarding ping6, yes I believe we should guard there too.

Reply:
> tsum is used to calculate the other variables, if it's 0 all of the rest will be 0 including the one passed to `ub32sqrtub16` which will end up crashing due to a division by zero.

Is it crash here?
https://github.com/apache/incubator-nuttx/blob/master/libs/libc/fixedmath/lib_ubsqrt.c#L60

Reply:
@Ouss4 https://github.com/apache/incubator-nuttx/pull/6124 may fix this issue too, but I think both change should merge.

Reply:
> Is it crash here?
https://github.com/apache/incubator-nuttx/blob/master/libs/libc/fixedmath/lib_ubsqrt.c#L60

Yes, because the function will receive 0 as argument.